### DISCUSSION TITLE
new o1 wrong

### DISCUSSION THREAD

**Discussion by 123hi123** [open-webui/open-webui#5703] [General]
# Bug Report

## Installation Method

docker

## Environment

- **Open WebUI Version:** v0.3.29
- **Ollama:** Not applicable

- **Operating System:** Windows 11
- **Browser:** Chrome 120.0

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of Open WebUI.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

The o1 model should be usable and respond to prompts within the Open WebUI interface.

## Actual Behavior:

The o1 model is not usable. When attempting to use it, an error message appears.
![image](https://github.com/user-attachments/assets/7509ceb3-984d-4680-a3af-f9f5bdf18053)

## Description

**Bug Summary:**
The o1 model is not functioning within the Open WebUI version 0.3.29 on Windows 11, preventing users from utilizing this specific model for text generation or conversation.

## Reproduction Details

**Steps to Reproduce:**
1. Launch Open WebUI v0.3.29 on Windows 11
2. Select the o1 model from the available models list
3. Enter a prompt in the chat interface
4. Click send or press enter to submit the prompt
5. Observe that no response is generated, or an error message appears

## Logs and Screenshots

**Browser Console Logs:**
[Please insert relevant browser console logs here]

**Screenshots/Screen Recordings (if applicable):**
[Please attach any relevant screenshots showing the issue with the o1 model]

## Additional Information

The issue appears to be specific to the o1 model in Open WebUI version 0.3.29 running on Windows 11. Other models within the Open WebUI may be functioning correctly. There are no visible error messages in the user interface, but the model simply fails to generate any output when prompted. This setup is not using Ollama.

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
@kivvi3412

**Comment by chris-sanders** [comment]
```
21:14:23 - LiteLLM:ERROR: main.py:414 - litellm.acompletion(): Exception occured - Error code: 400 - {'error': {'message': "Unsupported value: 'stream' does not support true with this model. Only the default (false) value is supported.", 'type': 'invalid_request_error', 'param': 'stream', 'code': 'unsupported_value'}}
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/litellm/llms/openai.py", line 1218, in async_streaming
    headers, response = await self.make_openai_chat_completion_request(
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litellm/llms/openai.py", line 785, in make_openai_chat_completion_request
    raise e
  File "/usr/local/lib/python3.11/site-packages/litellm/llms/openai.py", line 780, in make_openai_chat_completion_request
    response = await openai_aclient.chat.completions.create(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/openai/resources/chat/completions.py", line 1214, in create
    return await self._post(
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/openai/_base_client.py", line 1790, in post
    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/openai/_base_client.py", line 1493, in request
    return await self._request(
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/openai/_base_client.py", line 1584, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.BadRequestError: Error code: 400 - {'error': {'message': "Unsupported value: 'stream' does not support true with this model. Only the default (false) value is supported.", 'type': 'invalid_request_error', 'param': 'stream', 'code': 'unsupported_value'}}
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/litellm/main.py", line 393, in acompletion
    response = await init_response
               ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litellm/llms/openai.py", line 1271, in async_streaming
    raise OpenAIError(status_code=e.status_code, message=str(e))
litellm.llms.openai.OpenAIError: Error code: 400 - {'error': {'message': "Unsupported value: 'stream' does not support true with this model. Only the default (false) value is supported.", 'type': 'invalid_request_error', 'param': 'stream', 'code': 'unsupported_value'}}
```

I just came here when I saw Litellm error on o1 as well. I don't know if this is relevant but the API responded back to litellm that stream has to be false. If that's being set to true could be a problem here as well.

**Reply by kivvi3412** [reply]
> ```
> 21:14:23 - LiteLLM:ERROR: main.py:414 - litellm.acompletion(): Exception occured - Error code: 400 - {'error': {'message': "Unsupported value: 'stream' does not support true with this model. Only the default (false) value is supported.", 'type': 'invalid_request_error', 'param': 'stream', 'code': 'unsupported_value'}}
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.11/site-packages/litellm/llms/openai.py", line 1218, in async_streaming
>     headers, response = await self.make_openai_chat_completion_request(
>                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/litellm/llms/openai.py", line 785, in make_openai_chat_completion_request
>     raise e
>   File "/usr/local/lib/python3.11/site-packages/litellm/llms/openai.py", line 780, in make_openai_chat_completion_request
>     response = await openai_aclient.chat.completions.create(
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/openai/resources/chat/completions.py", line 1214, in create
>     return await self._post(
>            ^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/openai/_base_client.py", line 1790, in post
>     return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/openai/_base_client.py", line 1493, in request
>     return await self._request(
>            ^^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/openai/_base_client.py", line 1584, in _request
>     raise self._make_status_error_from_response(err.response) from None
> openai.BadRequestError: Error code: 400 - {'error': {'message': "Unsupported value: 'stream' does not support true with this model. Only the default (false) value is supported.", 'type': 'invalid_request_error', 'param': 'stream', 'code': 'unsupported_value'}}
> During handling of the above exception, another exception occurred:
> Traceback (most recent call last):
>   File "/usr/local/lib/python3.11/site-packages/litellm/main.py", line 393, in acompletion
>     response = await init_response
>                ^^^^^^^^^^^^^^^^^^^
>   File "/usr/local/lib/python3.11/site-packages/litellm/llms/openai.py", line 1271, in async_streaming
>     raise OpenAIError(status_code=e.status_code, message=str(e))
> litellm.llms.openai.OpenAIError: Error code: 400 - {'error': {'message': "Unsupported value: 'stream' does not support true with this model. Only the default (false) value is supported.", 'type': 'invalid_request_error', 'param': 'stream', 'code': 'unsupported_value'}}
> ```
> 
> I just came here when I saw Litellm error on o1 as well. I don't know if this is relevant but the API responded back to litellm that stream has to be false. If that's being set to true could be a problem here as well.

- Hi, according to your error message, O1 does not support streaming. You need to manually update the stream to "off" in the advanced parameters, as shown in the image below. See if that can make it work.
   <img width="919" alt="image" src="https://github.com/user-attachments/assets/d170105c-f903-4bb2-804e-396f850e472b">

**Reply by 123hi123** [reply]
I do that, actually i can use o1 before while i off stream but not it seems like a new problem

**Comment by chris-sanders** [comment]
Actually, after updating LiteLLM it works through LiteLLM. So you can try that, for now and check out how they're calling o1 models.

**Comment by kivvi3412** [comment]
> # Bug Report
> ## Installation Method
> docker
> 
> ## Environment
> * **Open WebUI Version:** v0.3.29
> * **Ollama:** Not applicable
> * **Operating System:** Windows 11
> * **Browser:** Chrome 120.0
> 
> **Confirmation:**
> 
> * [x]  I have read and followed all the instructions provided in the README.md.
> * [x]  I am on the latest version of Open WebUI.
> * [x]  I have included the browser console logs.
> * [x]  I have included the Docker container logs.
> * [x]  I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.
> 
> ## Expected Behavior:
> The o1 model should be usable and respond to prompts within the Open WebUI interface.
> 
> ## Actual Behavior:
> The o1 model is not usable. When attempting to use it, an error message appears. ![image](https://private-user-images.githubusercontent.com/36217117/370796129-7509ceb3-984d-4680-a3af-f9f5bdf18053.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjczMDk4MTgsIm5iZiI6MTcyNzMwOTUxOCwicGF0aCI6Ii8zNjIxNzExNy8zNzA3OTYxMjktNzUwOWNlYjMtOTg0ZC00NjgwLWEzYWYtZjlmNWJkZjE4MDUzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI2VDAwMTE1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVmMDhkM2I0NzQwNmRiMGQ3YzBjNDU0MmRkM2EzYmU4ODAyYjFjNTQ4MjM5NTk3MGEwNGEzMzBmZDcxNzIxYmEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zF2_mq486-eDT8ZN5zyq9mka3R0NpCAYZy3n4s9y8Eg)
> 
> ## Description
> **Bug Summary:** The o1 model is not functioning within the Open WebUI version 0.3.29 on Windows 11, preventing users from utilizing this specific model for text generation or conversation.
> 
> ## Reproduction Details
> **Steps to Reproduce:**
> 
> 1. Launch Open WebUI v0.3.29 on Windows 11
> 2. Select the o1 model from the available models list
> 3. Enter a prompt in the chat interface
> 4. Click send or press enter to submit the prompt
> 5. Observe that no response is generated, or an error message appears
> 
> ## Logs and Screenshots
> **Browser Console Logs:** [Please insert relevant browser console logs here]
> 
> **Screenshots/Screen Recordings (if applicable):** [Please attach any relevant screenshots showing the issue with the o1 model]
> 
> ## Additional Information
> The issue appears to be specific to the o1 model in Open WebUI version 0.3.29 running on Windows 11. Other models within the Open WebUI may be functioning correctly. There are no visible error messages in the user interface, but the model simply fails to generate any output when prompted. This setup is not using Ollama.
> 
> ## Note
> If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

- Hi, temporarily set the max token to the default value. The current project has not yet adapted the max token for o1.
- Reference [Why was max_tokens changed to max_completion_tokens?](https://community.openai.com/t/why-was-max-tokens-changed-to-max-completion-tokens/938077)
   
<img width="798" alt="image" src="https://github.com/user-attachments/assets/47b15e86-643c-46a1-b60e-dd88413a1f99">

**Comment by kivvi3412** [comment]
@123hi123 #5710

**Reply by 123hi123** [reply]
thanks a lot
Should I change to dev branch or this change will on the docker version

**Reply by kivvi3412** [reply]
- It will be merged into the main branch soon. You can check for new updates, and then update the Docker container accordingly.
   <img width="753" alt="image" src="https://github.com/user-attachments/assets/0337316e-df79-4bcb-a775-52dea5f529d9">

**Reply by 123hi123** [reply]
Thanks a lot
I have last curiousty about the news you posted 
It seems like openai have changed the parameters 13 days before but actually I happened this wrong yesterday.
I wondered how this happened

**Reply by kivvi3412** [reply]
> Thanks a lot I have last curiousty about the news you posted It seems like openai have changed the parameters 13 days before but actually I happened this wrong yesterday. I wondered how this happened

- If you are using a third-party API service provider in mainland China
- Originally the third-party API service provider chose to make their backend compatible with the O1 parameter, allowing you to use it without errors. 
- However, now the concurrency of O1 is too low (they may have been blocked upstream because OpenAI does not allow traffic from mainland China), and some service providers with insufficient upstream concurrency have chosen to use new parameters. 
- This causes errors when you use the old parameters, thus reducing the pressure on their servers.

**Reply by kivvi3412** [reply]
> Thanks a lot I have last curiousty about the news you posted It seems like openai have changed the parameters 13 days before but actually I happened this wrong yesterday. I wondered how this happened

- Current domestic service providers are all API service providers layered upon other API service providers, using panels like [one-api](https://github.com/songquanpeng/one-api) to set up servers with one click. Their recent backend updates might have caused errors for you.

**Reply by 123hi123** [reply]
Actually I am using the openai key for o1 model
But yeah, I use new api the convert it, and yesterday I change it to directly use in openwebui cause I can't read files, the read file function is ok but the function of o1 is dead 
That's all the story I have

**Comment by IA386** [comment]
Just wanted to add that I'm experiencing the same issue as described above with the `o3-mini` model in Open WebUI v0.5.10.  I've noticed this error occurs when the `max_tokens` setting is changed from the default value. The exact error is:

```
400: Unsupported parameter: 'max_tokens' is not supported with this model. Use 'max_completion_tokens' instead.
```

**Reply by kivvi3412** [reply]
```
    # Fix: O1 does not support the "max_tokens" parameter, Modify "max_tokens" to "max_completion_tokens"
    is_o1 = payload["model"].lower().startswith("o1-")
    if is_o1:
        payload = openai_o1_handler(payload)
    elif "api.openai.com" not in url:
        # Remove "max_completion_tokens" from the payload for backward compatibility
        if "max_completion_tokens" in payload:
            payload["max_tokens"] = payload["max_completion_tokens"]
            del payload["max_completion_tokens"]

    if "max_tokens" in payload and "max_completion_tokens" in payload:
        del payload["max_tokens"]

    # Convert the modified body back to JSON
    payload = json.dumps(payload)
```

Because the code logic judged that `o1-` uses the latest `max_completion_tokens`. Many people use third-party API providers, and old models like `GPT-4/3` can only use `max_tokens`. Therefore, for compatibility considerations, the parameters were modified separately for `o1-`, and adding `o3-` to the code allows it to be used.

**Reply by 123hi123** [reply]
> ```
>     # Fix: O1 does not support the "max_tokens" parameter, Modify "max_tokens" to "max_completion_tokens"
>     is_o1 = payload["model"].lower().startswith("o1-")
>     if is_o1:
>         payload = openai_o1_handler(payload)
>     elif "api.openai.com" not in url:
>         # Remove "max_completion_tokens" from the payload for backward compatibility
>         if "max_completion_tokens" in payload:
>             payload["max_tokens"] = payload["max_completion_tokens"]
>             del payload["max_completion_tokens"]
> 
>     if "max_tokens" in payload and "max_completion_tokens" in payload:
>         del payload["max_tokens"]
> 
>     # Convert the modified body back to JSON
>     payload = json.dumps(payload)
> ```
> 
> Because the code logic judged that `o1-` uses the latest `max_completion_tokens`. Many people use third-party API providers, and old models like `GPT-4/3` can only use `max_tokens`. Therefore, for compatibility considerations, the parameters were modified separately for `o1-`, and adding `o3-` to the code allows it to be used.

Interesting, I can use o3 direct in owu

**Comment by fancybear-dev** [comment]
This issue is also present for Azure AI Foundry OpenAI compatible endpoints for GPT-5 models.

fastapi.exceptions.HTTPException: 400: Unsupported parameter: 'max_tokens' is not supported with this model. Use 'max_completion_tokens' instead.
